packages/graphql/lib/src/core/observable_query.dart

```diff
@@ -273,9 +276,24 @@ class ObservableQuery<TParsed> {
   /// result that [QueryResult.isConcrete],
   /// handling the resolution of [lifecycle] from
   /// [QueryLifecycle.sideEffectsBlocking] to [QueryLifecycle.completed]
-  /// as appropriate
-  void onData(Iterable<OnData<TParsed>> callbacks) =>
-      _onDataCallbacks.addAll(callbacks);
+  /// as appropriate, unless if [removeAfterInvocation] is set to false.
+  ///
+  /// Returns a function for removing the added callbacks
+  void Function() onData(
+    Iterable<OnData<TParsed>> callbacks, [
+    bool removeAfterInvocation = true,
+  ]) {
```

Comment:
```suggestion
  void Function() onData(
    Iterable<OnData<TParsed>> callbacks, {
    bool removeAfterInvocation = true,
  }) {
```

Why not just use this syntax?

Reply:
I agree